src/test/java/com/ss/training/lms/LibrarianControllerTest.java

```diff
@@ -0,0 +1,207 @@
+package com.ss.training.lms;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.ss.training.lms.controller.LibrarianController;
+import com.ss.training.lms.entity.Book;
+import com.ss.training.lms.entity.BookCopies;
+import com.ss.training.lms.entity.LibraryBranch;
+import com.ss.training.lms.service.LibrarianService;
+
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+
+@SpringBootTest
+@TestInstance(Lifecycle.PER_CLASS)
+public class LibrarianControllerTest {
+
+	private MockMvc mockMvc;
+	
+	@Mock
+    LibrarianService librarianService;
+	
+	@InjectMocks
+	private LibrarianController librarianController;
+	
+	@BeforeAll
+	public void setup() throws Exception
+	{
+		mockMvc = MockMvcBuilders.standaloneSetup(librarianController).build();
+	}
```

Comment:
I am not sure how all this works at the moment but looks like you put in a ton of work! great stuff.